### PR TITLE
[Go] Document logs

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/4373

Fixes https://github.com/open-telemetry/opentelemetry.io/issues/4461

Preview: https://deploy-preview-4463--opentelemetry.netlify.app/docs/languages/go/instrumentation/#logs

This is documented similarly to https://opentelemetry.io/docs/languages/java/instrumentation/#logs as agreed here: https://github.com/open-telemetry/opentelemetry.io/issues/4373#issuecomment-2079977640

### CODE REVIEW

File: content/en/docs/languages/go/instrumentation.md [hunk 1]
```diff
@@ -876,7 +875,160 @@ meterProvider := metric.NewMeterProvider(
 
 ## Logs
 
-The logs API is currently unstable, documentation TBA.
+Logs are distinct from metrics and traces in that **there is no user-facing
+OpenTelemetry logs API**. Instead, there is tooling to bridge logs from existing
+popular log packages (e.g. slog, logrus, zap, logr) into the OpenTelemetry
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:880]:
```suggestion
popular log packages (such as slog, logrus, zap, logr) into the OpenTelemetry
```

File: content/en/docs/languages/go/instrumentation.md [hunk 2]
```diff
@@ -876,7 +875,158 @@ meterProvider := metric.NewMeterProvider(
...
+ecosystem. For rationale behind this design decision, see
+[Logging specification](/docs/specs/otel/logs/).
+
+The two typical workflows discussed below each cater to different application
+requirements.
+
+### Direct to collector (Experimental)
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:887]:
```suggestion
### Direct to Collector (experimental)
```

File: content/en/docs/languages/go/instrumentation.md [hunk 3]
```diff
@@ -876,7 +875,158 @@ meterProvider := metric.NewMeterProvider(
...
+
+In the direct to collector workflow, logs are emitted directly from an
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:889]:
```suggestion
In the direct-to-Collector workflow, logs are emitted directly from an
```

File: content/en/docs/languages/go/instrumentation.md [hunk 4]
```diff
@@ -876,7 +875,158 @@ meterProvider := metric.NewMeterProvider(
...
+suitable for all applications.
+
+To use this workflow:
+
+- Configure the OpenTelemetry [Log SDK](#logs-sdk-experimental) to export log
+  records to desired target destination (the
+  [collector][opentelemetry collector] or other).
+- Use an appropriate [Log Bridge](#log-bridge-experimental).
+
+#### Logs SDK (Experimental)
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:904]:
```suggestion
#### Logs SDK (experimental)
```

File: content/en/docs/languages/go/instrumentation.md [hunk 5]
```diff
@@ -876,7 +875,158 @@ meterProvider := metric.NewMeterProvider(
...
+		log.WithProcessor(processor),
+	)
+	return provider, nil
+}
+```
+
+Now that a `LoggerProvider` is configured, you can use it to set up a
+[Log Bridge](#log-bridge-experimental).
+
+#### Log Bridge (Experimental)
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:1007]:
```suggestion
#### Log Bridge (experimental)
```

File: content/en/docs/languages/go/instrumentation.md [hunk 6]
```diff
@@ -876,7 +875,158 @@ meterProvider := metric.NewMeterProvider(
...
+
+A log bridge is a component that bridges logs from an existing log package into
+the OpenTelemetry [Log SDK](#logs-sdk-experimental) using the [Logs Bridge
+API][logs bridge API]. Log bridges are available for various popular Go log
+packages:
+
+- [slog bridge][`go.opentelemetry.io/contrib/bridges/otelslog`]
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:1014]: Is there more than one? If you only want to give this one as an example, I'd suggest not putting it in a list, or at least adding "for example:".

Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:1014]:
```suggestion
- [slog bridge][otelslog]
```

File: content/en/docs/languages/go/instrumentation.md [hunk 7]
```diff
@@ -887,11 +1037,21 @@ telemetry backends.
 [opentelemetry specification]: /docs/specs/otel/
 [trace semantic conventions]: /docs/specs/semconv/general/trace/
 [instrumentation library]: ../libraries/
+[opentelemetry collector]:
+  https://github.com/open-telemetry/opentelemetry-collector
+[logs bridge API]: /docs/specs/otel/logs/bridge-api
+[log data model]: /docs/specs/otel/logs/data-model
+[`go.opentelemetry.io/contrib/bridges/otelslog`]:
+  https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelslog
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:1045]:
```suggestion
[otelslog]: https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelslog
```

**Review by chalin — COMMENTED** [review]
Oh, this reads well so far! I make a quick pass. See inline for preliminary copyedits.

File: content/en/docs/languages/go/instrumentation.md (hunk 6 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md:1014]: There will be more 😉 

E.g.
- https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5356
- https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5357

I wanted to preserve the consistency with Java.

File: content/en/docs/languages/go/instrumentation.md (hunk 1 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md:880]: Same is in Java. Should I fix it there as well?

File: content/en/docs/languages/go/instrumentation.md (hunk 2 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md:887]: Same is in other places. Should I fix it as well?

File: content/en/docs/languages/go/instrumentation.md (hunk 3 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md:889]: Same is in Java. Should I fix it there as well?

File: content/en/docs/languages/go/instrumentation.md (hunk 2 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md:887]: I am not sure if this is a good proposal. E.g.
- https://opentelemetry.io/docs/specs/semconv/resource/
- https://opentelemetry.io/docs/languages/go/exporters/

File: content/en/docs/languages/go/instrumentation.md (hunk 3 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md:889]: Shouldn't the header be also named "direct-to-Collector", then?

File: content/en/docs/languages/go/instrumentation.md (hunk 1 above)
Comment (svrnm) [reply content/en/docs/languages/go/instrumentation.md:880]: yes, if you want to, but in a separate PR.

File: content/en/docs/languages/go/instrumentation.md (hunk 3 above)
Comment (svrnm) [reply content/en/docs/languages/go/instrumentation.md:889]: Yes, there should be something consistent

**Review by svrnm — APPROVED** [review]
lgtm overall when @chalin's comments are addressed

File: content/en/docs/languages/go/instrumentation.md (hunk 2 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md]: I created https://github.com/open-telemetry/opentelemetry.io/issues/4488 to track this.

File: content/en/docs/languages/go/instrumentation.md (hunk 4 above)
Comment (chalin) [reply content/en/docs/languages/go/instrumentation.md:904]: Same comment as above:

- Add an explicit has ID (**here and for other headings**), for example: `{#logs-sdk}`
- Move the status mention into the section prose, and then you can link to the spec definition of the term, for example:

  ```markdown
  #### Logs SDK
  
  **Status**: [Experimental](/docs/specs/otel/document-status/)
  ```

  Or only keep the "Status" in the top-level "Direct-to-Collector" section's opening paragraph?

File: content/en/docs/languages/go/instrumentation.md [hunk 8]
```diff
@@ -876,7 +875,159 @@ meterProvider := metric.NewMeterProvider(
...
+Logs are distinct from metrics and traces in that **there is no user-facing
+OpenTelemetry logs API**. Instead, there is tooling to bridge logs from existing
+popular log packages (such as slog, logrus, zap, logr) into the OpenTelemetry
+ecosystem. For rationale behind this design decision, see
+[Logging specification](/docs/specs/otel/logs/).
+
+The two typical workflows discussed below each cater to different application
+requirements.
+
+### Direct-to-Collector (Experimental)
```
Comment (chalin) [inline content/en/docs/languages/go/instrumentation.md:887]: Btw, once this section looses its experimental status, the hash IDs to the section will become invalid. I have two alternative suggestions:

1. Move the status mention into the section prose, and then you can link to the spec definition of the term, for example:

    ```markdown
    ### Direct-to-Collector
    
    **Status**: [Experimental](/docs/specs/otel/document-status/)
    ```

2. Add an explicit has ID (**here and for other headings**), for example: `{#direct-to-collector}`

File: content/en/docs/languages/go/instrumentation.md (hunk 4 above)
Comment (pellared) [reply content/en/docs/languages/go/instrumentation.md]: > only keep the "Status" in the top-level "Direct-to-Collector" section's opening paragraph?

I like this. See: https://github.com/open-telemetry/opentelemetry.io/pull/4463/commits/6f3bc5ec289df4d07aa1504e24a10f4a91324c5b

**Comment by pellared** [comment]
PTAL @open-telemetry/go-approvers

File: content/en/docs/languages/go/instrumentation.md (hunk 4 above)
Comment (chalin) [reply content/en/docs/languages/go/instrumentation.md]: Yes, perfect, if that what you prefer. 👍🏻

**Pull request merged**